src/daft-session/src/session.rs

```diff
@@ -263,6 +357,28 @@ impl SessionState {
             _ => panic!("ambiguous table identifier"),
         }
     }
+
+    #[cfg(feature = "python")]
+    pub fn get_function(&self, name: &str) -> Result<Option<WrappedUDFClass>> {
```

Comment:
Did you consider using a Function trait in daft-catalog with a FunctionRef? We have this pattern already for Catalog and Table which hides all python things based upon the trait impl.

Reply:
yeah, but we only have 1 concrete udf type. We can refactor if/when we have non python based udfs.